### DISCUSSION TITLE
Question on the compatability for mobile builds like Android and IOS.

### DISCUSSION THREAD

**Discussion by dudetru25** [simonoliver/UnityFigmaBridge#34] [Q&A] [answered]
I haven't checked but would you say that your current shader to render a lot of these UI elements, are they more performant than unitys current solution especially when it comes to mobile devices? 

UI is one of the many things we are trying to optimize for as currently the way unity renders UI is inefficient. Noticed that the UI Toolkit isn't supported and there is a lot is custom class/shader work here so I would like to know if it's improved vs unitys current implementation.

**Comment by simonoliver** [comment, marked as answer]
The package uses Unity's UI system so you'll see similar performance but batching may be less efficient depending on what you are doing (esp for large numbers of elements). The custom shader/object (FigmaImage) adds native support for features that Unity's UI doesn't have (such as gradients, roundedness, shapes, outlines) to prevent the shapes having to be rendered to textures. I've built some projects for mobile which have worked fine, but for example rendering a gradient in the shader will not be as fast as rendering the gradient to a texture, so if you are targetting low end devices, you'll prob be better off avoiding procedural rendering.

UI toolkit is great and super performant but doesn't have support for enough rendering features to support full Figma export yet - support for custom shaders is supposed to be coming though.
